pkg/api/types.go

```diff
@@ -367,6 +373,28 @@ type GCEPersistentDiskVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// AWSPersistentDiskVolumeSource represents a Persistent Disk resource in AWS.
+//
+// An AWS EBS disk must exist and be formatted before mounting to a container.
+// The disk must also be in the same AWS zone as the kubelet.
+// A AWS EBS disk can only be mounted as read/write once.
+type AWSPersistentDiskVolumeSource struct {
+	// Unique name of the PD resource. Used to identify the disk in AWS
+	PDName string `json:"pdName"`
```

Comment:
Similarly, the AWS API has "volumeId", not PDName.

I see why you were thinking of 1 "cloud disk" for re-use across both GCE and AWS.  But so long as they must be separate implementations, I think the names of fields can map cleanly and obviously to the AWS API as opposed to mapping to the GCE API.  My $0.02.